core-tests/shared/src/test/scala/zio/autowire/AutoWireSpec.scala

```diff
@@ -212,6 +213,22 @@ object AutoWireSpec extends ZIOBaseSpec {
               s1 <- ZIO.service[String].provideLayer(tree)
               s2 <- ZIO.service[String].provideLayer(mermaid)
             } yield assertTrue(s1 == "Hello 1!", s2 == "Hello 2!")
+          },
+          test("takes trace from the implicit scope") {
+            var numTraces: Int = 0
+            val layer = {
+              implicit def trace: Trace = {
+                numTraces += 1
+                Trace.empty
+              }
+
+              ZLayer.make[String](
+                ZLayer.succeed(42),
+                ZLayer.fromFunction((_: Int).toString)
+              )
+            }
+
+            assertZIO(layer.build.as(numTraces))(equalTo(3))
```

Comment:
Thank you, at least now I won't have to remember why we had to do use explicit tags in the macros in the future 🥲

Reply:
This test was really hard to come up with 🤣